integration/truncindex_test.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+func genTruncIndex(normalName string) string {
+	return normalName[:(len(normalName)+1)/2]
+}
+
+func TestTruncIndex(t *testing.T) {
+	t.Logf("Pull an image")
+	const appImage = "busybox"
+	imgID, err := imageService.PullImage(&runtimeapi.ImageSpec{Image: appImage}, nil)
+	require.NoError(t, err)
+	imgTruncId := genTruncIndex(imgID)
+	defer func() {
+		assert.NoError(t, imageService.RemoveImage(&runtimeapi.ImageSpec{Image: imgTruncId}))
+	}()
+
+	t.Logf("Get image status by truncindex, truncId: %s", imgTruncId)
+	res, err := imageService.ImageStatus(&runtimeapi.ImageSpec{Image: imgTruncId})
+	require.NoError(t, err)
+	require.NotEqual(t, nil, res)
+	assert.Equal(t, imgID, res.Id)
+
+	// TODO(yanxuean): for failure test case where there are two images with the same truncindex.
+	// if you add n images at least two will share the same leading digit.
+	// "sha256:n" where n is the a number from 0-9 where two images have the same trunc,
+	// for example sha256:9
+	// https://github.com/kubernetes-incubator/cri-containerd/pull/352
+	// I am thinking how I get the two image which have same trunc.
+
+	// TODO(yanxuean): add test case for ListImages
+
+	t.Logf("Create a sandbox")
+	sbConfig := PodSandboxConfig("sandbox", "truncindex")
+	sb, err := runtimeService.RunPodSandbox(sbConfig)
+	require.NoError(t, err)
+	sbTruncIndex := genTruncIndex(sb)
+	var hasStopedSandbox bool
```

Comment:
s/Stoped/Stopped.

I won't block the PR for this. Will send another PR to fix this.